File: Core-backend-layer/Programming-fundamentals/Python/Threads/src/order_processor_thread.py
import threading

# Thread that extends the Thread class
class OrderProcessorThread(threading.Thread):
    def __init__(self, product, order_quantity):
        super().__init__()
        self.product = product
        self.order_quantity = order_quantity

    def run(self):
        # Simulate a process of processing an order
        self.product.decrease_quantity(self.order_quantity)

File: Core-backend-layer/Programming-fundamentals/Java/Threads/src/OrderProcessorThread.java
// Thread that extends the Thread class
class OrderProcessorThread extends Thread {
    private final Product product;
    private final int orderQuantity;

    public OrderProcessorThread(Product product, int orderQuantity) {
        this.product = product;
        this.orderQuantity = orderQuantity;
    }

    @Override
    public void run() {
        // Simulate a process of processing an order
        product.decreaseQuantity(orderQuantity);
    }
}